### PR TITLE
Global store support in smartsync plugin

### PR DESCRIPTION
**Pull request by wmathurin**
Also hooking up all new js tests
All tests passing

### CODE REVIEW

**Comment by wmathurin** [comment]
Will do changes discussed this morning in the next pull request

File: libs/SalesforceHybridSDK/HybridPluginTestAppTests/ForcetkTestSuite.m [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import <UIKit/UIKit.h>
+
+#import "ForcetkTestSuite.h"
+
+@implementation ForcetkTestSuite
+
+- (void)setUp
```
Comment (wmathurin) [inline libs/SalesforceHybridSDK/HybridPluginTestAppTests/ForcetkTestSuite.m]: Those tests were run from java for Android so I thought we should run them from Objective-C also

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartSyncPlugin/SFSmartSyncPlugin.m [hunk 2]
```diff
@@ -73,25 +83,27 @@ - (void)resetSyncManager
...
-- (void)handleSyncUpdate:(SFSyncState*)sync
+- (void)handleSyncUpdate:(SFSyncState*)sync isGlobal:(BOOL)isGlobal
 {
     dispatch_async(dispatch_get_main_queue(), ^{
         NSError *error = nil;
-        NSData *syncData = [NSJSONSerialization dataWithJSONObject:[sync asDict]
-                                                       options:0 // non-pretty printing
-                                                         error:&error];
+        NSMutableDictionary* detailDict = [NSMutableDictionary dictionaryWithDictionary:[sync asDict]];
+        detailDict[kSyncIsGlobalStoreArg] = isGlobal ? @YES : @NO;
```
Comment (wmathurin) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartSyncPlugin/SFSmartSyncPlugin.m]: Sync id is not enough, we need to let JS know whether it was the global sync manager or the other sync manager that fired the event

File: libs/SalesforceHybridSDK/HybridPluginTestAppTests/ForcetkTestSuite.m (hunk 1 above)
Comment (khawkins) [inline libs/SalesforceHybridSDK/HybridPluginTestAppTests/ForcetkTestSuite.m]: Oh yeah good idea.

**Comment by bhariharan** [comment]
:+1:

**Comment by khawkins** [comment]
Looks good.

**Pull request merged**